.github/workflows/test.yml

```diff
@@ -6,17 +6,12 @@ name: Tests / Code Coverage
 on:
   workflow_call:
 
-jobs:
-  cleanup-runs:
-    runs-on: ubuntu-latest
-    steps:
-      - uses: rokroskar/workflow-run-cleanup-action@master
-        env:
-          GITHUB_TOKEN: "${{ secrets.GITHUB_TOKEN }}"
-    if: |
-      !startsWith(github.ref, 'refs/tags/') &&
-      github.ref != 'refs/heads/master'
+concurrency:
+  group: ${{ github.workflow }}-${{ !startsWith(github.ref, 'refs/tags/') &&
+      github.ref != 'refs/heads/master' }}
```

Comment:
Thanks for working on this!

[question] why is the [example](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-only-cancel-in-progress-jobs-or-runs-for-the-current-workflow) here insufficient? Is `github.ref != 'refs/heads/master'` specified so that CI jobs on earlier commits on `master` aren't cancelled if new commits are added to master? This repo uses `main` instead of `master` as its canonical branch but preference to use the example if we don't need the `!startsWith(...` logic:

```
concurrency:
  group: ${{ github.workflow }}-${{ github.ref }}
  cancel-in-progress: true
```

Reply:
Yes, this logic ensures that `master` and `tags` branches are exempted from canceling  
 the earlier commits.

Reply:
Do you want me to update `master` to `main` or use the example as it is ?

```
concurrency:
  group: ${{ github.workflow }}-${{ github.ref }}
  cancel-in-progress: true
```

Reply:
Thanks for explaining. I think we can should `master` to `main`. Can you please also include a comment above these lines with a short explanation on the rationale?

```yaml
  # do not cancel jobs from earlier commits for tags or the main branch
  group: ${{ github.workflow }}-${{ !startsWith(github.ref, 'refs/tags/') &&
      github.ref != 'refs/heads/master' }}
```

Then I think this is ready to merge 👍

Reply:
Got it. Thanks for the review.

Reply:
Updated

Reply:
awesome thanks!